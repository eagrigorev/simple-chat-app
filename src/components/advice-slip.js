import { Button } from "@material-ui/core";
import { theme } from "./theme";
import { useDispatch, useSelector } from "react-redux";
import { getAdviceSlip } from "../store/selectors/advice-slip-selectors";
import { fetchAdviceSlipQuote } from "../store/actions/advice-slip-actions";
import { ADVICE_SLIP_DESCRIPTION } from "../const";
export const AdviceSlip = () => {
    const { quote, status } = useSelector(getAdviceSlip);
    const dispatch = useDispatch();
    const handleQuote = () => {
        dispatch(fetchAdviceSlipQuote());
    };
    if (status === "loading") {
        return (
            <div>
                <h2>AdviceSlip</h2>
                <p className="home">{ADVICE_SLIP_DESCRIPTION}</p>
                <p className="quote__message quote__message--loading">
                    Loading...
                </p>
            </div>
        );
    }
    return (
        <div>
            <h2>AdviceSlip</h2>
            <p className="home">{ADVICE_SLIP_DESCRIPTION}</p>
            {status !== "error" ? (
                <>
                    <p>Random Quote:</p>
                    <p className="quote__message quote__message--ok">{quote}</p>
                </>
            ) : (
                <p className="quote__message quote__message--error">
                    Something happened. Try again.
                </p>
            )}
            <Button
                className="quote__button"
                type="submit"
                variant="outlined"
                onClick={handleQuote}
                style={{ backgroundColor: theme.palette.secondary.main }}
            >
                Get a Quote
            </Button>
        </div>
    );
};
