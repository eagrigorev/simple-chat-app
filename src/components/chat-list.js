import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import { Link } from "react-router-dom";
import { useSelector } from "react-redux";
import { getChatList } from "../store/selectors/chat-list-selectors";
export const ChatList = (props) => {
    const chatList = useSelector(getChatList);
    return chatList.map((message) => (
        <div key={chatList.indexOf(message)}>
            <List>
                <ListItem>
                    <Link to={`/chats/${message.chatListId}`}>
                        Chat #{message.chatListId}
                    </Link>
                </ListItem>
                <ListItem>
                    {props.secondary
                        ? `Participants: ${message.chatListName}, ${message.chatListBot}`
                        : null}
                </ListItem>
            </List>
        </div>
    ));
};
