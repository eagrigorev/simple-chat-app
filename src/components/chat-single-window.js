import { useParams, Redirect } from "react-router-dom";
import { useSelector } from "react-redux";
import { getChatList } from "../store/selectors/chat-list-selectors";
import { getMessageList } from "../store/selectors/message-list-selectors";
export const ChatSingleWindow = () => {
    const messageList = useSelector(getMessageList);
    const chatList = useSelector(getChatList);
    let { chatId } = useParams();
    if (chatId > chatList.length) {
        return <Redirect to="/error" />;
    }
    const userById = chatList.find(
        (user) => user.chatListId === Number(chatId)
    );
    const chatFiltered = messageList.filter(
        (message) =>
            message.messageListName === userById.chatListName ||
            message.messageListBody ===
                `This is a bot reply to ${userById.chatListName}. That message was generated by Firebase in three seconds.`
    );
    return chatFiltered.map((message) => (
        <div className="message" key={chatFiltered.indexOf(message)}>
            {message.messageListName}: {message.messageListBody}
        </div>
    ));
};
