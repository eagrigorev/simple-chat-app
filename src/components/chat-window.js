import { useSelector } from "react-redux";
import "../firebase";
import { getMessageList } from "../store/selectors/message-list-selectors";
export const ChatWindow = () => {
    const messageList = useSelector(getMessageList);
    return messageList.map((message) => (
        <div className="message" key={messageList.indexOf(message)}>
            {message.messageListName}: {message.messageListBody}
        </div>
    ));
};
