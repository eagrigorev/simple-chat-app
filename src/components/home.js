export const Home = () => {
    return (
        <div>
            <h2>React Messaging App Prototype</h2>
            <p className="home">
                That chatting app was made with React and intended to highlight
                various tools of the framework. It has a built-in Google
                Firebase backend, Redux, Persist, password protection, third
                party API, login-register functionality, and so on.
            </p>
            <p>
                That project aimed to emulate a real conversation between users
                with a delayed message receiving mechanism, which was
                implemented using a middleware bot as a remote user.
            </p>
        </div>
    );
};
