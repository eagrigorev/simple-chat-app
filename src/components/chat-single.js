import { ChatSingleWindow } from "./chat-single-window";
export const ChatSingle = () => {
    return (
        <>
            <h2>Chat History</h2>
            <div className="chats__wrap">
                <ChatSingleWindow />
            </div>
        </>
    );
};
