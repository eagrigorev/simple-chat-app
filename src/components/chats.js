import React from "react";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Checkbox from "@material-ui/core/Checkbox";
import { ChatList } from "./chat-list";
import { ChatWindow } from "./chat-window";
import { theme } from "./theme";
export const Chats = (props) => {
    return (
        <div className="chats">
            <div>
                <h2>Latest Messages</h2>
                <div className="chats__wrap">
                    <ChatWindow />
                </div>
                <div className="form">
                    <TextField
                        required
                        className="outlined-required"
                        label="Name"
                        defaultValue=""
                        placeholder="John Doe"
                        variant="outlined"
                        autoFocus
                        name="name"
                        onChange={props.handleMsgNameChange}
                    />
                    <TextField
                        required
                        className="outlined-required"
                        label="Message"
                        defaultValue=""
                        placeholder="Hello, world!"
                        variant="outlined"
                        name="body"
                        onChange={props.handleMsgBodyChange}
                    />
                    <Button
                        type="submit"
                        variant="outlined"
                        onClick={props.updateMessageWithThunk}
                        style={{
                            backgroundColor: theme.palette.secondary.main,
                        }}
                    >
                        {" "}
                        Send
                    </Button>
                </div>
            </div>
            <div>
                <h2>Chats</h2>
                <div className="chats__list">
                    <ChatList secondary={props.secondary} />
                </div>
                <FormControlLabel
                    control={
                        <Checkbox
                            checked={props.secondary}
                            onChange={(event) =>
                                props.setSecondary(event.target.checked)
                            }
                        />
                    }
                    label="Toggle participants"
                />
            </div>
        </div>
    );
};
