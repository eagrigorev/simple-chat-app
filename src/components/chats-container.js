import { useSelector, useDispatch } from "react-redux";
import React, { useState, useEffect } from "react";
import "../firebase";
import firebase from "firebase";
import {
    messageNameChange,
    messageBodyChange,
} from "../store/actions/message-actions";
import {
    fbMessageListAddUserThunk,
    fbMessageListAddBotThunk,
    fbMessageListFetchStoreUpdate,
} from "../store/actions/message-list-actions";
import {
    fbChatListAddItemThunk,
    fbChatListFetchStoreUpdate,
} from "../store/actions/chat-list-actions";
import {
    fbGetChatId,
    fbSetChatIdThunk,
} from "../store/actions/chat-id-actions";
import { getMessageList } from "../store/selectors/message-list-selectors";
import { getMessage } from "../store/selectors/message-selectors";
import { getChatId } from "../store/selectors/chat-id-selectors";
import { Chats } from "./chats";
export const db = firebase.database();
export const ChatsContainer = () => {
    const dispatch = useDispatch();
    useEffect(() => {
        dispatch(fbMessageListFetchStoreUpdate());
        dispatch(fbGetChatId());
        dispatch(fbChatListFetchStoreUpdate());
    }, []);
    const { messageName, messageBody } = useSelector(getMessage);
    const messageList = useSelector(getMessageList);
    let { chatId } = useSelector(getChatId);
    const [secondary, setSecondary] = useState(false);
    const handleMsgNameChange = (event) => {
        dispatch(messageNameChange(event.target.value));
    };
    const handleMsgBodyChange = (event) => {
        dispatch(messageBodyChange(event.target.value));
    };
    const updateMessageWithThunk = () => {
        dispatch(fbMessageListAddUserThunk(messageName, messageBody));
        dispatch(fbMessageListAddBotThunk(messageName));
        const knownUser = messageList.find(
            (element) => element.messageListName === messageName
        );
        if (knownUser === undefined) {
            chatId++;
            dispatch(fbChatListAddItemThunk(chatId, messageName));
            dispatch(fbSetChatIdThunk(chatId));
        }
    };
    return (
        <Chats
            handleMsgNameChange={handleMsgNameChange}
            handleMsgBodyChange={handleMsgBodyChange}
            updateMessageWithThunk={updateMessageWithThunk}
            secondary={secondary}
            setSecondary={setSecondary}
        />
    );
};
