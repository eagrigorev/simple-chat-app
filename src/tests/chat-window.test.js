import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { ChatWindow } from "../components/chat-window";
import configureStore from "redux-mock-store";
describe("ChatWindow testing", () => {
    it("Returns a rendered component with a fake store", () => {
        const initialState = {
            messageList: [
                {
                    messageListName: "John Galt",
                    messageListBody: "Who am I?",
                },
            ],
        };
        const mockStore = configureStore()(initialState);
        render(
            <Provider store={mockStore}>
                <ChatWindow />
            </Provider>
        );
        const expectedComponent = "John Galt: Who am I?";
        expect(expectedComponent).toMatchSnapshot();
    });
});
