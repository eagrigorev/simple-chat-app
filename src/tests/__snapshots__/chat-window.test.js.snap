// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatWindow testing Returns a rendered component with a fake store 1`] = `"John Galt: Who am I?"`;
